enhancements/microshift/etcd-supportability.md

```diff
@@ -0,0 +1,249 @@
+are not always possible. It should therefore be as resilient as
+possible and take automated action when possible.
+
+There is a practical benefit of designing the workflows, processes,
+and tools for MicroShift so that an administrator’s experience with
+OCP can translate to experience with MicroShift. While we do not want
+MicroShift to diverge just for the sake of being different, there are
+already enough differences in operation due to the implementation
+details of MicroShift and the Device Edge product that some divergence
+from OCP may be inevitable.
+
+We do not want to force alignment at the expense of other design
+principles. For example, after recent issues compiling etcd into the
+same binary with the other aspects of MicroShift, it is now being
+delivered as a separate binary. However, to retain the design
+principle that the user should not have to orchestrate multiple
+services to use MicroShift, the MicroShift process manages the etcd
+process for the user. Starting and stopping MicroShift starts and
+stops etcd transparently. This is similar to the experience of using
+k3s or other single-binary distributions of kubernetes, even though
+the details are different.
+
+We need to keep the 4 footprints in mind (RAM, CPU, storage,
+bandwidth). Tools should be small, and included in the OS image along
+with MicroShift rather than delivered via container images at runtime.
+
+We have an aggressive schedule for bringing MicroShift to GA
+status. We therefore want to choose simple solutions, even if that
+means we need to do additional work to improve tools in later
+releases.
+
+### Workflow Description
+
+#### Database backup and restore
+
+For backup and restore we need to consider both the user-driven backup
+and the Greenboot backup scenarios.
+
+For the short term, we will start with copying the raw database
+files. This allows for a quick operation during Greenboot, including
+using copy-on-write backups, as well as user-driven backups using
+simple filesystem tools. For the longer term, we will wait for process
+improvements for etcd in OCP and potentially bring those into
+MicroShift later.
+
+#### Database defragmentation
+
+Because defragmentation is important for keeping the database size
+below the quota, we want to emphasize the use of automatic
+defragmentation rather than requiring administrators to trigger it
+explicitly.
+
+We will start with the implementation in
+https://github.com/openshift/microshift/pull/1395/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R241-R247,
+including the configuration parameters and their defaults:
+
+* Min size threshold: 100 MB
+* Max % fragmentation: 45%
+* Check period: 5 min
+* Startup defragmentation: bool, defaulting to true
+* DB size quota: 2GB
+
+#### Delivering etcdctl
+
+QE and support teams have asked for `etcdctl` for working with the
+etcd database. `sos` also supports it.
+
+We consider `etcdctl` as nice-to-have for 4.13 and a requirement for
```

Comment:
If microshift delivers a seamless UX when it comes to managing etcd, should the tool follow the same principle? I mean not having etcdctl but use that as a microshift command for homogeneity.

Reply:
Yes, good point, this should really say "features of etcdctl".

Reply:
If the `etcdctl` won't be present, then `sos` etcd plugin won't work correctly. 
https://github.com/sosreport/sos/blob/main/sos/report/plugins/etcd.py#L66

Also, plugin doesn't check if `etcdctl` is present, so most likely this fails on microshift.  Yeah, nothing will be executed, even the etcd plugin will fail due to the `/var/lib/etcd` not present.

Reply:
Since we're treating the etcd inside microshift as part of a unit, we will want to add some etcd capture to the microshift plugin. We may be able to do that in a way that reuses code in the existing plugin, if it makes sense.

Reply:
I've updated this section to talk about etcdtctl features.